pkg/util/helper/binding.go

```diff
@@ -153,12 +163,20 @@ func FetchWorkload(dynamicClient dynamic.Interface, restMapper meta.RESTMapper,
 // EnsureWork ensure Work to be created or updated.
 //nolint:gocyclo
 // Note: ignore the cyclomatic complexity issue to get gocyclo on board. Tracked by: https://github.com/karmada-io/karmada/issues/460
-func EnsureWork(c client.Client, workload *unstructured.Unstructured, clusterNames []string, overrideManager overridemanager.OverrideManager,
-	binding metav1.Object, scope apiextensionsv1.ResourceScope) error {
-	referenceRSP, desireReplicaInfos, err := calculateReplicasIfNeeded(c, workload, clusterNames)
-	if err != nil {
-		klog.Errorf("Failed to get ReplicaSchedulingPolicy for %s/%s/%s, err is: %v", workload.GetKind(), workload.GetNamespace(), workload.GetName(), err)
-		return err
+func EnsureWork(c client.Client, workload *unstructured.Unstructured, clusterNames []string, clusters []workv1alpha1.TargetCluster,
+	overrideManager overridemanager.OverrideManager, binding metav1.Object, scope apiextensionsv1.ResourceScope) error {
+	var desireReplicaInfos map[string]int64
+	var referenceRSP *v1alpha1.ReplicaSchedulingPolicy
+	var err error
+	isBindingReplicasReady := IsBindingReplicasReady(clusters)
```

Comment:
No need to check if resource without 'replicas' ?

Reply:
if resource is without replicas the replicas will be zero in clusters。 This judgement will return false .